### ISSUE TITLE
pynio causes dependency conflicts in py36 CI build

### ISSUE THREAD

**Issue by crusaderky** [pydata/xarray#3154]
On Saturday night, all Python 3.6 CI builds started failing. Python 3.7 is unaffected.
See https://dev.azure.com/xarray/xarray/_build/results?buildId=362&view=logs

MacOSX py36:
```
UnsatisfiableError: The following specifications were found to be in conflict:
  - pynio
  - python=3.6
  - rasterio
```

Linux py36:
```
UnsatisfiableError: The following specifications were found to be in conflict:
  - cfgrib[version='>=0.9.2']
  - h5netcdf
  - pynio
```

**Comment by HasanAhmadQ7** [comment]
I am getting a different error for linux py36

Linux py36
```
UnsatisfiableError: The following specifications were found to be in conflict:
  - cfgrib[version='>=0.9.2']
  - h5py
  - pynio
```

**Comment by shoyer** [comment]
I can reproduce this issue locally:
```
$ conda create -n xarray-tmp -c conda-forge python=3 pynio h5py
Collecting package metadata (current_repodata.json): done
Solving environment: failed with current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:



Package hdfeos5 conflicts for:
pynio -> hdfeos5==5.1.15=0
Package zlib conflicts for:
pynio -> zlib
python=3 -> zlib[version='1.2.*|1.2.11|1.2.8|>=1.2.11,<1.3.0a0']
Package hdfeos2 conflicts for:
pynio -> hdfeos2
Package libgfortran conflicts for:
pynio -> libgfortran[version='>=3.0,<4.0.0.a0|>=3.0.1,<4.0.0.a0']
Package libuuid conflicts for:
pynio -> libuuid[version='>=2.32.1,<3.0a0']
Package hdf5 conflicts for:
pynio -> hdf5[version='1.10.1|1.8.15.1|>=1.10.2,<1.10.3.0a0|>=1.10.4,<1.10.5.0a0|>=1.10.5,<1.10.6.0a0']
h5py -> hdf5[version='1.10.1|1.10.1.*|1.10.2.*|1.8.15.*|1.8.17.*|1.10.2.*,>=1.10.2,<1.10.3.0a0|1.8.17|1.8.17.*|1.8.18|1.8.18.*|>=1.10.2,<1.10.3.0a0|>=1.10.3,<1.10.4.0a0|>=1.10.4,<1.10.5.0a0|>=1.10.4,<1.10.5.0a0|>=1.10.4,<1.10.5.0a0',build='mpi_mpich_*|mpi_openmpi_*']
Package jpeg conflicts for:
pynio -> jpeg
Package python conflicts for:
pynio -> python[version='2.7.*|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
h5py -> python[version='2.7.*|3.4.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
Package proj4 conflicts for:
pynio -> proj4[version='4.9.3|>=4.9.3,<4.9.4.0a0|>=5.2.0,<5.2.1.0a0|>=6.0.0,<6.0.1.0a0|>=6.1.0,<6.1.1.0a0']
Package libgcc conflicts for:
pynio -> libgcc
Package krb5 conflicts for:
pynio -> krb5
Package mpich conflicts for:
h5py -> mpich[version='>=3.2,<3.3.0a0']
Package curl conflicts for:
pynio -> curl
Package openmpi conflicts for:
h5py -> openmpi[version='>=3.1,<3.2.0a0']
Package readline conflicts for:
python=3 -> readline[version='6.2.*|7.0|>=7.0,<8.0a0|>=8.0,<9.0a0']
Package numpy conflicts for:
h5py -> numpy[version='1.10.*|1.11.*|1.12.*|1.13.*|>=1.8|>=1.8,<1.14|>=1.9.3,<2.0a0']
pynio -> numpy[version='1.10.*|1.11.*|1.12.*|1.13.*|1.9.*|>=1.11|>=1.14|>=1.7']
Package pip conflicts for:
python=3 -> pip
Package mpi4py conflicts for:
h5py -> mpi4py
Package ncurses conflicts for:
python=3 -> ncurses[version='>=6.1,<6.2.0a0']
Package libgdal conflicts for:
pynio -> libgdal[version='2.1.*|2.2.*|>=2.2.4,<2.3.0a0|>=2.4.0,<2.5.0a0|>=2.4.1,<2.5.0a0']
Package libiconv conflicts for:
pynio -> libiconv
Package tk conflicts for:
python=3 -> tk[version='8.5.*|8.6.*|>=8.6.8,<8.7.0a0|>=8.6.9,<8.7.0a0']
Package libcxx conflicts for:
python=3 -> libcxx[version='>=4.0.1']
pynio -> libcxx[version='>=4.0.1']
Package hdf4 conflicts for:
pynio -> hdf4[version='>=4.2.13,<4.3.0a0']
Package jasper conflicts for:
pynio -> jasper[version='>=1.900.1,<2.0a0']
Package xz conflicts for:
python=3 -> xz[version='5.0.*|5.2.*|>=5.2.3,<5.3.0a0|>=5.2.4,<5.3.0a0']
Package libpng conflicts for:
pynio -> libpng
Package bzip2 conflicts for:
python=3 -> bzip2[version='>=1.0.6,<2.0a0']
Package sqlite conflicts for:
python=3 -> sqlite[version='3.13.*|3.20.*|3.9.*|>=3.24.0,<4.0a0|>=3.25.1,<4.0a0|>=3.25.2,<4.0a0|>=3.25.3,<4.0a0|>=3.26.0,<4.0a0|>=3.28.0,<4.0a0']
Package six conflicts for:
h5py -> six
Package libnetcdf conflicts for:
pynio -> libnetcdf[version='4.4.*|>=4.5|>=4.6.1,<4.7.0a0|>=4.6.2,<4.6.3.0a0|>=4.6.2,<4.7.0a0']
Package libffi conflicts for:
python=3 -> libffi[version='>=3.2.1,<3.3.0a0']
Package openssl conflicts for:
python=3 -> openssl[version='1.0.*|>=1.0.2o,<1.0.3a|>=1.0.2p,<1.0.3a|>=1.1.1a,<1.1.2a']
Package unittest2 conflicts for:
h5py -> unittest2
Package gdal conflicts for:
pynio -> gdal[version='1.11.*|2.1.*|2.2.*|1.11.4']
Package g2clib conflicts for:
pynio -> g2clib[version='1.5.*|1.6.*']
Note that strict channel priority may have removed packages required for satisfiability.
```

Maybe hdf5 is the problem? h5py seems to want the mpi version, but pynio doesn't.

Perhaps one of our friendly conda-forge experts can help diagnose the issue? @ocefpaf any ideas?

**Comment by akrherz** [comment]
I am not an expert, but will chime in.  Verify that you have `conda-forge` before the `defaults` channel and that you have `strict` channel priority enabled.

**Comment by shoyer** [comment]
> Verify that you have `conda-forge` before the `defaults` channel and that you have `strict` channel priority enabled.

I'm not 100% sure what this looks like, but we use `conda env` to make our build environments and only list `conda-forge` under channels.

**Comment by ocefpaf** [comment]
The problem here is that we had to remove some old builds of `gdal` (a `pynio` dependency) due to an incompatibility with a newer `icu`. However, the new builds are using `hdf5 1.10.5` but `h5py` is not ready for it yet. (`h5py` still uses `hdf 1.10.4`.)

**Comment by shoyer** [comment]
OK, this is fixed for now by removing `pynio` from the CI builds.

**Comment by max-sixty** [comment]
Should we close? Or wait for the dependencies to update?

**Comment by shoyer** [comment]
I renamed the issue -- let's leave it open for now until we can restore pynio.

**Comment by crusaderky** [comment]
Closed by #3340
